libs/feature/editor/src/lib/models/editor-config.model.ts

```diff
@@ -1,29 +1,56 @@
-import { FormFieldConfig } from '../components/record-form/form-field'
+import { CatalogRecordKeys } from '@geonetwork-ui/common/domain/model/record'
 
 // Expressions should be enclosed in `${}` to be recognized as such
 // eg. ${dateNow()}
 export type EditorFieldExpression = `$\{${string}}`
 
-export interface EditorFieldConfig {
+export type EditorFieldValue = string | number | boolean | unknown
+
+export interface FormFieldConfig {
+  labelKey: string
+  hintKey?: string
+  tooltipKey?: string
+  required?: boolean
+  locked?: boolean
+  invalid?: boolean
+  invalidHintKey?: string
+}
+
+export interface EditorField {
   // configuration of the form field used as presentation; optional, nothing shown if not defined
-  formFieldConfig?: FormFieldConfig
+  formFieldConfig: FormFieldConfig
 
   // name of the target field in the record; will not change the record directly if not defined
-  model?: string
+  model?: CatalogRecordKeys
 
   // a hidden field won't show but can still be used to modify the record
   // FIXME: currently this is redundant with an absence of formFieldConfig but necessary for clarity
   hidden?: boolean
 
   // the result of this expression will replace the field value on save
   onSaveProcess?: EditorFieldExpression
+
+  value?: EditorFieldValue
 }
 
-export type EditorFieldsConfig = EditorFieldConfig[]
+export interface EditorSection {
+  label: string
+  description: string
+  hidden: boolean
+  fields: EditorField[]
+}
 
-export type EditorFieldValue = string | number | boolean | unknown
+export interface EditorFieldPage {
+  index: number
+  label: string
```

Comment:
Properties such as this one should be called `labelKey` (as in the fields), because they are meant to hold translation keys, which is actually the case in the rest of your PR :+1:

Reply:
Also I would leave the "label" and "description" keys optional, instead of having to set them to an empty string for the title or description to be hidden